components/formats-gpl/src/loci/formats/in/CellH5Reader.java

```diff
@@ -577,7 +580,7 @@ private void parseROIs(int s) {
             cellObjectName + " " + objectLabelId, roiManagerRoiIndex);
 
           String shapeID =
-            MetadataTools.createLSID("Shape", roiManagerRoiIndex, 0);
+            MetadataTools.createLSID(cellObjectName, roiManagerRoiIndex, 0);
```

Comment:
This is unfortunately going to cause the OME-XML to be invalid - the ID must be of the format ```Shape:*:*```.

It might be better to call ```store.setRectangleText(cellObjectName, roiManagerRoiIndex, 0)``` in the reader, and then update ROIHandler to set the ImageJ Roi name to ```shapeObject.getText()``` on line 308 (falling back to ```shapeObject.getID()``` if the text is null).  That way the OME-XML remains valid, but the descriptive name shows up in the ROI manager.